lib/devices/gadgets/fitness_equipment.dart

```diff
@@ -413,8 +413,11 @@ class FitnessEquipment extends DeviceBase {
         "ensuring that manufacturer name ($manufacturerName) contains manufacturer prefix ${descriptor!.manufacturerPrefix}",
       );
     }
-    return manufacturerName?.toLowerCase().contains(descriptor!.manufacturerPrefix.toLowerCase()) ??
-        false || descriptor!.manufacturerPrefix == "Unknown";
+    if (manufacturerName == null) {
```

Comment:
I like the breakdown. However it's not 100% equivalent to the previous version. If `descriptor!.manufacturerPrefix == "Unknown"` but `manufacturerName` is null. I guess that can happen when even a 2nd read is not successful. I had some problems in the past, that's why I have the 2nd try for read. I can see two ways: 1. Return true immediately if `descriptor!.manufacturerPrefix == "Unknown"`, and then have the other predicate, in that case the `??` can remain. The other way is maybe to use a magic value other than the null for the initial value of `manufacturerName`. `null` indicates that the the reading wasn't attempted yet, and once it failed there wouldn't be consecutive attempts. Maybe we can just have it as a non nullable variable with an empty string initial value. If we assume that no machine will report empty string value (I think we can assume that) we can just get rid of the whole nullability. I'm not sure if that'll affect other files of the app. Now that I dig deeper this whole clause was like that even before your change (both with exclamation marks), so we can also decide to defer further changes after your PR (although that's somewhat dangerous because maybe I'll forget it).